snuba/migrate.py

```diff
@@ -41,11 +46,15 @@ def get_schema():
         logger.warn(difference)
 
 
-def run(conn, dataset):
-    schemas = []
-    if dataset.get_table_writer():
-        schemas.append(dataset.get_table_writer().get_schema())
-    schemas.append(dataset.get_dataset_schemas().get_read_schema())
+def run(conn: Client, dataset: Dataset) -> None:
+    schemas: Sequence[Schema] = []
+    write_storage = dataset.get_writable_storage()
+    if write_storage:
+        writer = write_storage.get_table_writer()
+        if writer:
+            schemas.append(writer.get_schema())
```

Comment:
It seems like that some read schemas are just same as the write schemas. Does this mean we are running that schema migration twice?

Reply:
You mentioned in https://github.com/getsentry/snuba/pull/828#discussion_r391814858 that DDL shouldn't care about which storages are read or write, just migrate everything retuned by `get_all_storages()`

Reply:
yes and it is already happening since DatasetSchemas lists the same schema twice for read and write. We should fix this but it is ok now since all migrations are idempotent. I think the new migration system should enforce this.

Reply:
yeah, I will change this, I just added the indentation without changing the logic.

Reply:
Actually, datasets are still aware of their schemas right. And storages can share schemas? Why go through storage rather than get the schemas with `dataset.get_all_schemas()`. It seems migrations only need to know all the schemas, doesn't really need to know anything about storage.

Reply:
yes we could do that. It would require a refactoring of StorageSchemas that I would keep for a separate PR because it would be quite verbose. Moreover such refactoring is needed anyway since there is no reason for StorageSchema to know which storages are for read and which ones are for writes.

Reply:
Though returning the list of schemas from the dataset would mean that such method would have to decide which is the order those schemas would have to be applied for migrations, which does not seem right, because that method is not generic anymore and contextualized on migrations. 
A better alternative seems adding a dependency relation between storage so that we know which storage should be migrated first (and set one schema per storage). How does it sound ?

Reply:
Good point, you mean like a global list of storages in the order that they should be bootstraped/migrated?

Reply:
not sure whether it should be global or per dataset but yes.